cartconditions/Coupon.php

```diff
@@ -168,6 +168,9 @@ protected function validateCoupon($couponModel)
         if ($couponModel->hasReachedMaxRedemption())
             throw new ApplicationException(lang('igniter.cart::default.alert_coupon_maximum_reached'));
 
+        if ($couponModel->customer_redemptions AND !$user)
+            throw new ApplicationException(lang('igniter.coupons::default.alert_coupon_login_required'));
+
```

Comment:
It's safe to remove the redundant check on line 174

Reply:
Good spot, forgot to remove it